server/router/auth.ts

```diff
@@ -18,22 +18,58 @@ const validIDs = [
   '322014396101-8u88pc3q00v6dre4doa64psr9349bhum.apps.googleusercontent.com',
 ];
 
-async function verify(clientID: string, token: string): Promise<void> {
+async function verify(clientID: string, token: string): Promise<LoginTicket> {
   const client = new OAuth2Client(clientID);
-  client.verifyIdToken({
+  const authRes = await client.verifyIdToken({
     idToken: token,
     audience: validIDs,
-  })
-    .then((ticket) => ticket.getPayload())
-    .catch((err) => { throw err; });
+  });
+  return authRes;
 }
 
 // Verify an authentication token
-router.post('/', async (req, res) => {
-  const { token, clientID } = req.body;
-  verify(token, clientID)
-    .then(() => res.send({ success: true }))
-    .catch(() => res.send({ success: false }));
+router.post('/', (req, res) => {
+  const {
+    token, clientID, userType, email,
+  } = req.body;
+  verify(clientID, token)
+    .then((authRes) => {
+      const payload = authRes.getPayload();
+      if (payload && payload.aud === clientID) {
+        const params = {
+          TableName: userType === 'rider' ? 'Riders' : 'Drivers',
+          ProjectionExpression: 'id, email',
+          FilterExpression: 'email = :user_email',
+          ExpressionAttributeValues: {
+            ':user_email': email,
+          },
+        };
+        docClient.scan(params, (err, data) => {
+          if (err) {
+            res.send(err);
+          } else {
+            const userList = data.Items;
+            if (typeof userList![0] === 'undefined') {
+              res.send({
+                success: true,
```

Comment:
I'm wondering if 'success' and 'exists' are necessary now. I feel like the information that they convey can be inferred from the existence or non-existence of the 'id'. If we allow 'id' to be null, we can eliminate the need for 'success' and 'exists' because if 'id' is null, then 'success' and 'exists' must be false. This works in the non-error cases.

The only place where 'success' might be necessary is in the error cases. This will allow us to distinguish between the cases that are actually errors and those cases in which someone simply does not have an account yet.

Change the responses sent by this function as you see fit, but aim for uniformity among them.

Reply:
The highlighted code section is not exactly what I wanted to show, but I hope that my explanation conveys the point.

Reply:
that makes sense, ill change it